data_model/src/permissions.rs

```diff
@@ -11,19 +11,54 @@ use alloc::{
 #[cfg(feature = "std")]
 use std::collections::{btree_map, btree_set};
 
-use derive_more::Display;
+use derive_more::{Constructor, Display, FromStr};
 use getset::Getters;
 #[cfg(feature = "ffi")]
 use iroha_ffi::{ffi_export, IntoFfi, TryFromFfi};
 use iroha_schema::IntoSchema;
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
-use crate::{Name, Value};
+use crate::{Identifiable, Name, Registered, Value};
 
 /// Collection of [`PermissionToken`]s
 pub type Permissions = btree_set::BTreeSet<PermissionToken>;
 
+/// Identification of a permission token.
+#[derive(
+    Debug,
+    Display,
+    Constructor,
+    FromStr,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Decode,
+    Encode,
+    Deserialize,
+    Serialize,
+    IntoSchema,
+)]
+pub struct Id {
+    /// Permission token name
+    pub name: Name,
```

Comment:
It's what we will register in domain, so it needs a wrapper to implement `Registered`.
Alternative would be to implement `Registered` with `With = Name` for `PermissionToken`, but I think it doesn't fit `Registered` semantics well, since we aren't registering `PermissionToken` itself.
Another alternative would be something like `PermissionTokenDefinition` analogous to `AssetDefinition`.

Reply:
> Another alternative would be something like PermissionTokenDefinition analogous to AssetDefinition.

Sounds good to me. This also opens the possibility of registering a custom validator for that permission token.

Reply:
> Sounds good to me

Sure, I could re-implement it that way.

Reply:
Yeah, I also like the `PermissionTokenDefinition` idea. This will allow to specify required parameters